main.tf

```
@@ -308,3 +308,49 @@ resource "aws_codepipeline" "default" {
     }
   }
 }
+
+resource "random_string" "webhook_secret" {
+  count  = local.enabled && var.webhook_enabled ? 1 : 0
+  length = 32
+
+  # Special characters are not allowed in webhook secret (AWS silently ignores webhook callbacks)
+  special = false
+}
+
+locals {
+  webhook_secret = join("", random_string.webhook_secret.*.result)
+  webhook_url    = join("", aws_codepipeline_webhook.webhook.*.url)
+}
+
+resource "aws_codepipeline_webhook" "webhook" {
+  count           = local.enabled && var.webhook_enabled ? 1 : 0
```

move this to `locals` https://github.com/cloudposse/terraform-aws-cicd/blob/master/main.tf#L7
```hcl
webhook_count = local.webhook_enabled ? 1 : 0
```
```suggestion
  count           = local. webhook_count
```